frontend/src/components/profile/info/BadgeCabinetDialog.tsx

```diff
@@ -0,0 +1,190 @@
+import CloseIcon from '@mui/icons-material/Close';
+import {
+    Box,
+    Card,
+    Dialog,
+    DialogContent,
+    DialogTitle,
+    FormControl,
+    IconButton,
+    MenuItem,
+    Select,
+    Stack,
+    Typography,
+} from '@mui/material';
+import { Badge, BadgeType } from './badgeHandler';
+import BadgeProgress from './BadgeProgress';
+import CustomBadge from './CustomBadge';
+
+interface BadgeCabinetDialogProps {
+    isOpen: boolean;
+    onClose: () => void;
+    badgeCategory: string;
+    setBadgeCategory: (category: string) => void;
+    allBadges: Badge[];
+    allGradBadges: Badge[];
+    handleBadgeClick: (badge: Badge) => void;
```

Comment:
I don't think we need `handleBadgeClick` in this dialog. Opening dialogs on top of dialogs is not a good UI, and this dialog will already show the badge image, the description and the requirements to get it, so there is no reason to open the other dialog from this one.